### PR TITLE
Add seeders/leechers to notifiers and cache. Prepare SR for "manual snatch" feature

### PR DESCRIPTION
**Pull request by fernandog**
- Adds seeder/leechers in notifiers message
- Adds seeders/leechers in the "Downloading..." INFO log message - more visible to users
- Prepare cache.db to show seeders/leechers when feature **"manual snatch"** is implemented so users can choose releases based on seeds/leechers
- Update cache.db when parsing items at DailySearcher so users have updated seeders/leechers when manual snatch

**Next step** for another PR: create a pop up showing cache.db results for that show so user can manually snatch a result

http://feathub.com/SickRage/SickRage/+23

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [ ] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by miigotu** [comment]
No.

**Comment by fernandog** [comment]
which number?

**Comment by miigotu** [comment]
Cool so lets take something I said no about and mix some other features in and make it worse...

**Comment by fernandog** [comment]
@miigotu
Are you sure they are not related? Think again:

manual snatch feature need cached seeders/leechers to avoid ddos providers. or are we going to hit providers for every manual snatch?

And remember that feature request is the #1 FR. So what is the real reason for closing this as I did all changes as @labrys said?

So instead of saying 'no', what is you solution to make it work and/or add that FR? Or why not 'do not merge' until more improvents to code?

I bet you and @SickRage/collaborators that wont find issues in my code. Say your bet.

**Comment by miigotu** [comment]
I can point out at least 5 bugs in this PR, besides the fact that you are trying to force your seeders/leechers in notifications.

You need to learn to accept criticism, how to take no for an answer and move on, and how to manage risk.

idgaf if you PR it 10 times, the answer will be the same when you just keep PR the same shit.

**Comment by fernandog** [comment]
I fixed the issues pointed in the other PR
I not seeing you pointing issues in the code.
Just empty criticism

As you already checked where are those 5 bugs? Improvements dont count.

File: sickrage/providers/torrent/TorrentProvider.py [hunk 1]
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
     def _get_storage_dir(self):
         return sickbeard.TORRENT_DIR
 
+    def _get_result_info(self, item):
+        if isinstance(item, (dict, FeedParserDict)):
+            seeders = item.get('seeders', '')
```
Comment (miigotu) [inline sickrage/providers/torrent/TorrentProvider.py:102]: no bug?

File: sickrage/providers/torrent/TorrentProvider.py [hunk 2]
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
...
+            leechers = item.get('leechers', '')
```
Comment (miigotu) [inline sickrage/providers/torrent/TorrentProvider.py:103]: or here?

File: sickrage/providers/torrent/TorrentProvider.py [hunk 3]
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
...
+
+        elif isinstance(item, (list, tuple)) and len(item) > 1:
+            seeders = item[3]
+            leechers = item[4]
+        else:
+            seeders = ''
```
Comment (miigotu) [inline sickrage/providers/torrent/TorrentProvider.py:109]: or here?

File: sickrage/providers/torrent/TorrentProvider.py [hunk 4]
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
...
+            leechers = ''
```
Comment (miigotu) [inline sickrage/providers/torrent/TorrentProvider.py:110]: what about here?

File: sickrage/providers/GenericProvider.py [hunk 5]
```diff
@@ -443,6 +446,9 @@ def _get_size(self, item):  # pylint: disable=unused-argument,no-self-use
     def _get_storage_dir(self):  # pylint: disable=no-self-use
         return ''
 
+    def _get_result_info(self, item):  # pylint: disable=no-self-use
+        return '', ''
```
Comment (miigotu) [inline sickrage/providers/GenericProvider.py:450]: or this?

File: sickbeard/tvcache.py [hunk 6]
```diff
@@ -60,6 +60,14 @@ def __init__(self, providerName):
             if not self.hasColumn(providerName, 'version'):
                 self.addColumn(providerName, 'version', "NUMERIC", "-1")
 
+            # add seeders column to table if missing
+            if not self.hasColumn(providerName, 'seeders'):
+                self.addColumn(providerName, 'seeders', "NUMERIC", "")
```
Comment (miigotu) [inline sickbeard/tvcache.py:65]: here?

File: sickbeard/tvcache.py [hunk 7]
```diff
@@ -60,6 +60,14 @@ def __init__(self, providerName):
...
+
+            # add leechers column to table if missing
+            if not self.hasColumn(providerName, 'leechers'):
+                self.addColumn(providerName, 'leechers', "NUMERIC", "")
```
Comment (miigotu) [inline sickbeard/tvcache.py:69]: here?

File: sickbeard/search_queue.py [hunk 8]
```diff
@@ -274,7 +283,10 @@ def run(self):
             if searchResult:
                 for result in searchResult:
                     # just use the first result for now
-                    logger.log(u"Downloading " + result.name + " from " + result.provider.name)
+                    if result.seeders and result.leechers:
```
Comment (miigotu) [inline sickbeard/search_queue.py:286]: so when leechers is 0 dont show seeder count?

File: sickbeard/search_queue.py [hunk 9]
```diff
@@ -224,7 +230,10 @@ def run(self):
                 if searchResult:
                     for result in searchResult:
                         # just use the first result for now
-                        logger.log(u"Downloading " + result.name + " from " + result.provider.name)
+                        if result.seeders and result.leechers:
```
Comment (miigotu) [inline sickbeard/search_queue.py:233]: also here?

File: sickbeard/search_queue.py [hunk 10]
```diff
@@ -180,7 +183,10 @@ def run(self):
 
             if searchResult:
                 # just use the first result for now
-                logger.log(u"Downloading " + searchResult[0].name + " from " + searchResult[0].provider.name)
+                if searchResult[0].seeders and searchResult[0].leechers:
```
Comment (miigotu) [inline sickbeard/search_queue.py]: and here?

File: sickbeard/search_queue.py [hunk 11]
```diff
@@ -142,7 +142,10 @@ def run(self):
             else:
                 for result in foundResults:
                     # just use the first result for now
-                    logger.log(u"Downloading " + result.name + " from " + result.provider.name)
+                    if result.seeders and result.leechers:
```
Comment (miigotu) [inline sickbeard/search_queue.py]: and here?

File: sickbeard/search.py [hunk 12]
```diff
@@ -169,11 +169,14 @@ def snatchEpisode(result, endStatus=SNATCHED):
 
         if curEpObj.status not in Quality.DOWNLOADED:
             try:
-                notifiers.notify_snatch(curEpObj._format_pattern('%SN - %Sx%0E - %EN - %QN') + " from " + result.provider.name)
-            except:
+                if result.seeders and result.leechers:
```
Comment (miigotu) [inline sickbeard/search.py:172]: and here?

File: sickbeard/classes.py [hunk 13]
```diff
@@ -103,6 +103,12 @@ def __init__(self, episodes):
         # size of the release (-1 = n/a)
         self.size = -1
 
+        # seeders of the release
+        self.seeders = ""
```
Comment (miigotu) [inline sickbeard/classes.py:107]: so seeders and leechers is a string now?

File: sickbeard/classes.py [hunk 14]
```diff
@@ -103,6 +103,12 @@ def __init__(self, episodes):
...
+
+        # leechers of the release
+        self.leechers = ""
```
Comment (miigotu) [inline sickbeard/classes.py:110]: ?

**Comment by miigotu** [comment]
Talk some more shit?

**Comment by miigotu** [comment]
I dont give a shit if you fix every bit of what I point out. We arent making these ridiculous changes just to show seeds/leechers in a NOTIFICATION.

File: sickbeard/classes.py (hunk 13 above)
Comment (fernandog) [inline sickbeard/classes.py:107]: can't be int. cause 0 means 0 seeds. seeds = "" is for nzb

File: sickbeard/search.py (hunk 12 above)
Comment (fernandog) [inline sickbeard/search.py]: where?! what wrong?

File: sickbeard/search_queue.py (hunk 11 above)
Comment (fernandog) [inline sickbeard/search_queue.py]: where? what's wrong?

File: sickbeard/search_queue.py (hunk 10 above)
Comment (fernandog) [inline sickbeard/search_queue.py]: where? what's wrong?

File: sickbeard/search_queue.py (hunk 9 above)
Comment (fernandog) [inline sickbeard/search_queue.py:233]: multiple point the same code doesn't count as bug.
if seeders = "" means it's NZB. 0 is for zero seeds

File: sickbeard/search_queue.py (hunk 8 above)
Comment (fernandog) [inline sickbeard/search_queue.py:286]: easy fix. not something to close PR isn't?

File: sickbeard/tvcache.py (hunk 7 above)
Comment (fernandog) [inline sickbeard/tvcache.py:69]: as I said: empty = nzb.  0 seeds is zero seeds.

File: sickrage/providers/GenericProvider.py (hunk 5 above)
Comment (fernandog) [inline sickrage/providers/GenericProvider.py:450]: what is wrong?

NZB don't have extra info. what should be returned?

File: sickrage/providers/torrent/TorrentProvider.py (hunk 4 above)
Comment (fernandog) [inline sickrage/providers/torrent/TorrentProvider.py:110]: its your original code here! lol
made as @labrys said. make a specific function for it

get_title uses the same concept!

File: sickrage/providers/torrent/TorrentProvider.py (hunk 1 above)
Comment (fernandog) [inline sickrage/providers/torrent/TorrentProvider.py:102]: your original code in get_title

**Comment by fernandog** [comment]
you didn't read the PR

this is for notification **AND feature manual snatch feature**. The most wanted FR

as for "bugs" you "found", no bugs at all. just improvements.

File: sickbeard/classes.py (hunk 14 above)
Comment (fernandog) [inline sickbeard/classes.py:110]: what didnt you understand? I will explain

File: sickrage/providers/GenericProvider.py (hunk 5 above)
Comment (miigotu) [inline sickrage/providers/GenericProvider.py:450]: methods ALWAYS need to return the same type, regardless of what is subclassing it.

You could have saved yourself a whole lot of time and effort if you would have just accepted the no to begin with.  You could have saved me a whole lot of time and effort also, and let me waste that time and effort on more effective shit.

File: sickrage/providers/torrent/TorrentProvider.py (hunk 2 above)
Comment (fernandog) [inline sickrage/providers/torrent/TorrentProvider.py:103]: your original code in get_title

File: sickrage/providers/torrent/TorrentProvider.py (hunk 3 above)
Comment (fernandog) [inline sickrage/providers/torrent/TorrentProvider.py:109]: already explained

**Comment by miigotu** [comment]
So it is ok to mix 2 PR into one to try and sneak in something I SAID NO ABOUT? Even if the answer was yes mixing features is not ok. just stfu and stop spamming me now.

I swear that you could argue that the sun wont come up, and then the sun comes up and you will argue that it wont go down. Both ways you are wrong.

File: sickbeard/tvcache.py (hunk 6 above)
Comment (fernandog) [inline sickbeard/tvcache.py:65]: easy fix. not something to close PR isn't?

File: sickrage/providers/GenericProvider.py (hunk 5 above)
Comment (fernandog) [inline sickrage/providers/GenericProvider.py:450]: not something to close PR isn't?

File: sickbeard/search.py (hunk 12 above)
Comment (fernandog) [inline sickbeard/search.py:172]: easy fix. not something to close PR isn't?

**Comment by miigotu** [comment]
Every single one of those is a bug, not an improvement. The code is incorrect. The entire PR is not necessary at all to begin with.

Besides the fact that those arent the only problems here, those are just the first 15 or so that I found in 30 seconds.

**Comment by adaur** [comment]
I don't know if the code is good, but the idea is. We'll need to have the seeders and leechers in the cache to display them when one needs to choose its release to download.

**Comment by fernandog** [comment]
@adaur 
I removed the seeders/leechers from notifiers in my local branch and fixed all points mii said.

https://github.com/fernandog/SickRage/commit/0c98493461e28b90406c4e67bc93c65951c25b7f

**Comment by duramato** [comment]
@fernandog size also needs to be cached like seeder/leechers , right @adaur ?

**Comment by fernandog** [comment]
Added new commit that adds size
https://github.com/fernandog/SickRage/commits/seeds_in_notify

and rebased to last develop

**Pull request closed without merging**